File: string-to-integer-atoi/string-to-integer-atoi.py
class Solution:
    def myAtoi(self, s: str) -> int:
        sign=1
        result=0                                                                    
        s = s.strip()
        if s == "":
            return 0
        if s[0] == "+":
            s=s[1:]
            sign=1
        elif s[0] == "-":
            s=s[1:]
            sign=-1
        else:
            sign=1
        
        for character in s:
            if '0'<=character<='9':
                result = result*10 + int(character)
            else:
                break
                
        result=result*sign
        
        if result<-2147483648:
            return -2147483648
        elif result>2147483647:
            return 2147483647
        
        return result
        
    
    
         
                
        
        
            
        

File: string-to-integer-atoi/string-to-integer-atoi.java
class Solution {
    public int myAtoi(String s) {
        s = s.trim();
        if(s.length() == 0) {
            return 0;
        }
        long num = 0;
        
        boolean isNegative = s.charAt(0) == '-' ? true : false;
        for(int i=0; i< s.length(); i++){
            char c = s.charAt(i);
            if(i==0 && (c=='+' || c=='-')){
                continue;
            }
            if(Character.isDigit(c)){
                num = num* 10 + (c - '0');
                if(!isNegative && num>=Integer.MAX_VALUE){
                    return Integer.MAX_VALUE;
                }
                else if(isNegative && num>=(long)Integer.MAX_VALUE+1){
                    return Integer.MIN_VALUE;
                }
            }
            else{
                break;
             }
            }
        int res = (int) num;
        return isNegative ? -res: res;
        
    }
}